### PR TITLE
Issue/2056

### PR DESCRIPTION
**Pull request by DrXyzzy**
Ref: #2056 

- 'required' changed to '' (empty string) for create_project options 'title' and 'description'
- Test cases added to src/smc-hub/test/api/project.coffee for empty title / desc in create_project.
- Timeout increased in src/smc-hub/test/api/advanced.coffee for first code run in a project.

Full api test run:
```
~/smc/src/smc-hub$ mocha test/api
...
-- 65 passing (16s) --
```

### CODE REVIEW

**Comment by DrXyzzy** [comment]
Added doc & test for unlink_passport.

**Comment by DrXyzzy** [comment]
- Use API and not direct `api.db` call to publish path in text_files.coffee test.
- Add example of publishing a file to query doc at bottom of message.coffee.
- Rename advanced.coffee to project_exec.coffee in api tests.

File: src/smc-util/message.coffee [hunk 1]
```diff
@@ -572,11 +572,48 @@ message
...
+        strategy:
+            init  : required
+            desc  : 'passport strategy'
+        id:
+            init  : required
+            desc  : 'numeric id for user and passport strategy'
+    desc:"""
+Unlink a passport auth for the account.
+
+Strategy is one of: `facebook`, `github`, `google`, `twitter`.
```
Comment (haraldschilly) [inline src/smc-util/message.coffee]: Just nitpicking, but in practice the strategies are defined in the database. Maybe just list them as possible examples? The truth comes from `/auth/strategies`, i.e. look here https://cocalc.com/auth/strategies

Comment (DrXyzzy) [inline src/smc-util/message.coffee]: thanks I will update now

Comment (williamstein) [inline src/smc-util/message.coffee]: Yep, and there are way more than 4 already implemented.  I've been waiting for somebody to requests a fifth... Never happens.

Comment (DrXyzzy) [inline src/smc-util/message.coffee]: done

**Comment by DrXyzzy** [comment]
Auth strategies doc corrected per @haraldschilly suggestion.

**Pull request merged**